### ISSUE TITLE
windows/386 and windows/amd64 fail to build on tip of release-branch.go1.20

### ISSUE THREAD

**Issue by dmitshur** [golang/go#64453]
From https://build.golang.org/?branch=release-branch.go1.20:

<img width="272" alt="image" src="https://github.com/golang/go/assets/1924134/c82df06a-fa07-4cca-930f-b297eb3b9f69">

Starting with [CL 545356](https://go.dev/cl/545356), building the toolchain fails with:

```
Building Go cmd/dist using C:\workdir\go1.4. (go1.20.6 windows/amd64)
Building Go toolchain1 using C:\workdir\go1.4.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Exception 0xc0000005 0x8 0x0 0x0
PC=0x0

runtime.asmstdcall()
	C:/workdir/go/src/runtime/sys_windows_amd64.s:62 +0x5e fp=0x22fc90 sp=0x22fc80 pc=0x468dbe
rax     0x0
rbx     0xb562f8
rcx     0xba9800
rdi     0x7fffffdd000
rsi     0x22fe88
rbp     0x22fe58
rsp     0x22fc78
r8      0x0
r9      0x22fec8
r10     0xc0000041c0
r11     0xc0000041d0
r12     0x22fea8
r13     0xc0000041f0
r14     0xb54e80
r15     0x20
rip     0x0
rflags  0x10293
cs      0x33
fs      0x53
gs      0x2b
go tool dist: FAILED: C:\workdir\go\pkg\tool\windows_amd64\go_bootstrap install cmd/asm cmd/cgo cmd/compile cmd/link: exit status 2
```

_(source: https://build.golang.org/log/cd37d246fef65b77de9d481cbc2eca002ca6b0f1)_

This is also causing "build binary archive" step of the release process to fail for windows/386 and windows/amd64 ports, hence this is a release-blocker.

This is the tracking issue for this particular problem, which we need to resolve in some way to unblock the release process. If needed, I'll open a separate issue to figure out why this wasn't caught by the LUCI trybots, only the old infra builders, and to fix that. (CC @golang/release.)

CC @rolandshoemaker.

**Comment by dmitshur** [comment]
Two differences I'm spotting to note:

1. In old infra, go1.20.6 is being used as the bootstrap version. In LUCI, go1.17.13 is the bootstrap version, which is the minimum required for Go 1.20.

2. In old infra, windows-amd64-2008 builder is used, which is Windows Server 2008. In LUCI, I see "OS Version: 10.0.19045" which I think is Windows Server 2016.

**Comment by rolandshoemaker** [comment]
I believe 10.0.19045 is Windows 10 version 22H2. I believe ProcessPrng was added in Windows 10/Server 2016, so anything before that is going to be broken.

As far as I am aware we already dictate that Windows 10/Server 2016 is the lowest supported version (per Windows 10 and higher or Windows Server 2016), so breakage on the Server 2008 builder seems semi-expected (slightly surprising this hasn't popped up before).

**Comment by dmitshur** [comment]
Go 1.20 is the last release that still supports Windows 7 and up (https://go.dev/doc/go1.20#windows), whereas Go 1.21 started to require Windows 10/Windows Server 2016 and higher (https://go.dev/doc/go1.21#windows).

**Comment by rolandshoemaker** [comment]
Ah yes, we'll need to revert https://go.dev/cl/545356 then.

**Comment by dmitshur** [comment]
The revert was done in [CL 545995](https://go.dev/cl/545995).

This wasn't caught via LUCI trybots because we have only setup Windows 10, not older, in it so far. See [here](https://cs.opensource.google/go/x/build/+/luci-config:main.star;l=555-556;drc=26793f5e46222e6f85a4d8362356f018cf0f6f57). Given Go 1.20 will be out of support window in only a couple months, we can rely on the old builders to provide the signal, until that release ages out.
